logstash-core/src/main/java/org/logstash/plugins/AliasRegistry.java

```diff
@@ -138,8 +140,8 @@ private Map<PluginCoordinate, String> loadAliasesDefinitionsFromInputStream(Inpu
                 return Collections.emptyMap();
             }
 
-            // decode yaml to nested maps
-            final Map<String, Map<String, String>> aliasedDescriptions;
+            // decode yaml to Map<PluginType, List<AliasPlugin>> structure
```

Comment:
```suggestion
            // decode yaml to Map<String, List<AliasPlugin>> structure
```

Would be nice to move map's key type from `String` to  `PluginType` that could carry more context information, but maybe it's part of a follow up PR